src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsToken.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Text.Json.Serialization;
+
+namespace GSoft.Extensions.Http.Authentication.ClientCredentialsGrant;
+
+internal sealed class ClientCredentialsToken
+{
+    [JsonPropertyName("accessToken")]
+    public string AccessToken { get; init; } = string.Empty;
```

Comment:
Does having the setter as `init` impact the ability for the `AccessToken` to be refreshed? Or is this the client's `AccessToken` to obtain a jwt?

Reply:
Yep no refresh token. The `init` setter here is to make that POCO class immutable. Immutability in general programming is often associated with less chances of programming mistakes.

Reply:
Ok good so this is the client secret which will not be refreshed (and cannot). Wanted to be absolutely sure about that